sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -95,9 +95,10 @@ object DateTimeUtils {
     sdf
   }
 
-  def newDateFormat(formatString: String, timeZone: TimeZone): DateFormat = {
+  def newDateFormat(formatString: String, timeZone: TimeZone, isLenient: Boolean): DateFormat = {
     val sdf = new SimpleDateFormat(formatString, Locale.US)
     sdf.setTimeZone(timeZone)
+    sdf.setLenient(isLenient)
```

Comment:
we can test it with lenient false. this is  a util func, if test is ok, should we always set it to false?